src/ssl.c

```diff
@@ -4962,71 +4967,80 @@ int ProcessBuffer(WOLFSSL_CTX* ctx, const unsigned char* buff,
         switch (cert->keyOID) {
         #ifndef NO_RSA
             case RSAk:
+            #ifdef HAVE_PK_CALLBACKS
+                keyType = rsa_sa_algo;
+            #endif
+                /* Determine RSA key size by parsing public key */
+                idx = 0;
+                ret = wc_RsaPublicKeyDecode(cert->publicKey, &idx, NULL,
+                    cert->pubKeySize);
+                if (ret < 0)
+                    break;
+                keySz = ret;
```

Comment:
I don't see how this was resolved.  wc_RsaPublicKeyDecode() returns 0 on success, keySz is set to 0.  I don't see how that can ever be the correct keySz.